pkg/tools/mcp/oauth.go

```diff
@@ -336,6 +336,16 @@ type oauthTransport struct {
 }
 
 func (t *oauthTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	var bodyBytes []byte
+	if req.Body != nil && req.Body != http.NoBody {
+		var err error
+		bodyBytes, err = io.ReadAll(req.Body)
```

Comment:
Hi, @jba referred me here as I'm fixing this in the MCP client implementation.

I think you need to close the body after reading it, as RoundTripper says that RoundTrip must always close the body.